### PR TITLE
Secure SSH tool

### PR DESCRIPTION
**Pull request by abbashus**
### Description
This PR  aims to fix the following issues #29, #32  and introduces logging and control via CLI arguments.

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/project-website-search/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: tools/ssh-bastion-ec2/ssh-bastion-ec2 [hunk 1]
```diff
@@ -117,12 +119,12 @@ def ssh_ec2(remote_host, bastion_ip, key):
   retVal = 1
   try:
     tmp_key_path, ssh_cmd = _get_ssh_command(remote_host, bastion_ip, key)
-    # print("SSH_COMMAND: ", ssh_cmd)
-    retVal = os.system(ssh_cmd)
+    logging.info("SSH_COMMAND: %s", ' '.join(ssh_cmd))
```
Comment (AMoo-Miki) [inline tools/ssh-bastion-ec2/ssh-bastion-ec2:122]: Should this be `logger`? There is more below.

File: tools/ssh-bastion-ec2/ssh-bastion-ec2 [hunk 2]
```diff
@@ -117,12 +119,12 @@ def ssh_ec2(remote_host, bastion_ip, key):
...
+    retVal = subprocess.call(ssh_cmd)
   except Exception as e:
     # It is bad practice to except on any exception but we want to remove
     # the tmp ssh key regardless of what might happen.
-    print("ssh failed with exception: {}".format(e))
+    logging.exception("SSH failed with exception: ")
```
Comment (AMoo-Miki) [inline tools/ssh-bastion-ec2/ssh-bastion-ec2:127]: It says "with exception:" but `e` is not being stringified.

File: tools/ssh-bastion-ec2/ssh-bastion-ec2 [hunk 3]
```diff
@@ -163,27 +163,38 @@ def parse_arguments(argv):
...
   return random.choice(ip_list) if ip_list else None
 
 
 def __main__(argv):
   args = parse_arguments(argv)
+  initialize_logging(args.log_level)
   bastion_ip = get_random_bastion_ip(args.tag_name, args.tag_value)
   if not bastion_ip:
-    print("No bastion host IPs found. Exiting!")
+    logger.error("No bastion host IPs found. Exiting!")
```
Comment (AMoo-Miki) [inline tools/ssh-bastion-ec2/ssh-bastion-ec2:190]: I don't see the exiting happening here. Am I missing something?

File: tools/ssh-bastion-ec2/ssh-bastion-ec2 (hunk 2 above)
Comment (abbashus) [reply tools/ssh-bastion-ec2/ssh-bastion-ec2:127]: Why testing, the stringified execption was bring printed to the console without adding it explicitly to the exception.

File: tools/ssh-bastion-ec2/ssh-bastion-ec2 (hunk 1 above)
Comment (abbashus) [reply tools/ssh-bastion-ec2/ssh-bastion-ec2]: Done.

File: tools/ssh-bastion-ec2/ssh-bastion-ec2 (hunk 3 above)
Comment (abbashus) [reply tools/ssh-bastion-ec2/ssh-bastion-ec2:190]: Add appropriate exits.

**Review by AMoo-Miki — APPROVED** [review]
perhaps revisit adding the exception message in 127.

**Review by CEHENKLE — APPROVED** [review]

**Comment by abbashus** [comment]
> perhaps revisit adding the exception message in 127.

Verified again, exception need not be explicitly passed to `logging.excpetion` in exception context. The output is same with or without it. 

Also found a related [answer](https://stackoverflow.com/questions/5191830/how-do-i-log-a-python-error-with-debug-information?answertab=votes#tab-top) that confirms it.

**Pull request merged**